docs/grammar/Solidity.g4

```diff
@@ -189,20 +190,40 @@ locals[
 	(Semicolon | body=block);
 
 /**
- * Definitions of the special fallback and receive functions.
+ * Definitions of the special fallback function.
  */
-fallbackReceiveFunctionDefinition
+fallbackFunctionDefinition
 locals[
-	boolean visibilitySet = false,
 	boolean mutabilitySet = false,
+	boolean virtualSet = false,
+	boolean overrideSpecifierSet = false,
+	boolean hasParameters = false
+]
+:
+	kind=(Fallback | Receive) LParen (parameterList { $hasParameters = true; } )? RParen
```

Comment:
```suggestion
	kind=Fallback LParen (parameterList { $hasParameters = true; } )? RParen
```

Reply:
Also, `functionDefinition` above has `(Fallback | Receive)` too. Isn't that superfluous?

Reply:
For functionDefinition this is how you define `function fallback()` - a different thing.

Reply:
Oh, I see. "fallback" string will be recognized as a `Fallback` token rather than  `identifier` but we still want to allow it as a name...